src/api/Comments.js

```diff
@@ -40,6 +50,143 @@ class Comments extends OffsetBasedAPI {
 
         this.successCallback({ ...data, entries: comments });
     };
+
+    /**
+     * API URL for comments endpoint
+     *
+     * @param {string} [id] - A box comment file id
+     * @return {string} base url for comments
+     */
+    getCommentsUrl(id?: string): string {
+        const suffix: string = id ? `/${id}` : '';
+        return `${this.getBaseApiUrl()}/comments${suffix}`;
+    }
+
+    /**
+     * API for creating a comment on a file
+     *
+     * @param {BoxItem} file - File object for which we are creating a comment
+     * @param {string} message - Comment message
+     * @param {string} taggedMessage - Comment message with @mentions
+     * @param {Function} successCallback - Success callback
+     * @param {Function} errorCallback - Error callback
+     * @return {Promise}
+     */
+    createComment({
+        file,
+        message,
+        taggedMessage,
+        successCallback,
+        errorCallback
+    }: {
+        file: BoxItem,
+        message: string,
+        taggedMessage?: string,
+        successCallback: Function,
+        errorCallback: Function
+    }): Promise<void> {
+        const { id = '', permissions } = file;
```

Comment:
Why are you defaulting id?

Reply:
flow doesn't understand that my try catch will guarantee that ID is available by the time we get to line 99. So it thinks i'm possibly calling getTypedFileId on `undefined`. Trying to refactor around this